cli/cmd/stat.go

```diff
@@ -79,7 +79,7 @@ If no resource name is specified, displays stats about all resources of the spec
 func init() {
 	RootCmd.AddCommand(statCmd)
 	statCmd.PersistentFlags().StringVarP(&namespace, "namespace", "n", "default", "Namespace of the specified resource")
-	statCmd.PersistentFlags().StringVarP(&timeWindow, "time-window", "t", "1m", "Stat window (one of: \"10s\", \"1m\", \"10m\", \"1h\")")
+	statCmd.PersistentFlags().StringVarP(&timeWindow, "time-window", "t", "1m", "Stat window (for example: \"10s\", \"1m\", \"10m\", \"1h\")")
```

Comment:
Now that we're switching to an actual duration, you can use a duration flag here, and that will handle validation of the duration automatically.

```
	statCmd.PersistentFlags().DurationVarP(&timeWindow, "time-window", "t", time.Minute, "Stat window (for example: \"10s\", \"1m\", \"10m\", \"1h\")")
```

And in the event that the flag isn't a valid duration, that returns:

```
$ bin/go-run cli stat -t asdf
Error: invalid argument "asdf" for "-t, --time-window" flag: time: invalid duration asdf
```

Reply:
I did consider this approach. The drawback here is that we lose information about the original string when it's parsed into a `time.Duration`. This issue compounds because Prometheus accepts a subset of valid time strings, specifically `[integer][h|m|s]`. Passing in `1m` would yield `1m0s` which fails in Prometheus.

We could go a step further and try to guess at the best representation:
https://play.golang.org/p/5N4bjnEucoA
... this would still yield queries that don't exactly match the user's input. Another option would be to convert everything to seconds.

The approach I took in this PR is to use `time.Parse` to validate the input is a valid time string, but then to still pass the user's original string through to Prometheus. This also ensures the `timeWindow` string in the API response matches what the user typed. Let me know what you think.

Reply:
Ah yep, that all makes sense to me. Thanks for the explanation.